barretenberg/cpp/src/barretenberg/api/bbapi_ultra_honk.cpp

```diff
@@ -0,0 +1,272 @@
+#include "barretenberg/api/bbapi_ultra_honk.hpp"
+#include "barretenberg/api/bbapi_shared.hpp"
+#include "barretenberg/circuit_checker/circuit_checker.hpp"
+#include "barretenberg/common/serialize.hpp"
+#include "barretenberg/common/throw_or_abort.hpp"
+#include "barretenberg/dsl/acir_format/acir_format.hpp"
+#include "barretenberg/dsl/acir_format/acir_to_constraint_buf.hpp"
+#include "barretenberg/dsl/acir_format/serde/witness_stack.hpp"
+#include "barretenberg/flavor/mega_flavor.hpp"
+#include "barretenberg/honk/types/aggregation_object_type.hpp"
+#include "barretenberg/ultra_honk/decider_proving_key.hpp"
+#include "barretenberg/ultra_honk/ultra_prover.hpp"
+#include "barretenberg/ultra_honk/ultra_verifier.hpp"
+#include <iomanip>
+#include <sstream>
+
+namespace bb::bbapi {
+
+CircuitProve::Response CircuitProve::execute(const BBApiRequest& request) &&
+{
+    auto constraint_system = acir_format::circuit_buf_to_acir_format(std::move(circuit.bytecode));
+    auto witness = acir_format::witness_buf_to_witness_data(std::move(this->witness));
+
+    acir_format::AcirProgram program{ constraint_system, witness };
+
+    // Determine honk recursion level based on settings
+    uint32_t honk_recursion = settings.honk_recursion;
+    if (honk_recursion == 0 && settings.recursive) {
```

Comment:
oof is this really right? could be. Lucas is the only one who really knows what the hell these flags mean